File: src/main/webapp/app/customize/order.service.js
/**
 * Created by jeyrschabu on 9/1/16.
 */

'use strict';

function OrderService($http, API_PREFIX) {
    var orderService = this;

    function placeOrder(data) {
        return $http({
            url: API_PREFIX + '/order',
            method: 'POST',
            data: data
        });
    }

    orderService.placeOrder = placeOrder;
}

File: src/main/java/com/rooftopcoder/web/services/OrderService.java
package com.rooftopcoder.web.services;

import com.rooftopcoder.web.data.MongoConnectionConfig;
import com.rooftopcoder.web.data.providers.ModelProvider;
import com.rooftopcoder.web.models.Order;
import com.rooftopcoder.web.models.Spec;
import sun.reflect.generics.reflectiveObjects.NotImplementedException;

/**
 * Created by jeyrschabu on 9/1/16.
 */
public class OrderService extends BaseService<Order> {
    private static final String INITIAL_DATA = "";

    public OrderService setDataProvider(MongoConnectionConfig config) {
        super.setDataProvider(config, Order.class);
        return this;
    }

    protected String getInitialJsonData() {
        return INITIAL_DATA;
    }

    protected ModelProvider<Order> getProvider() {
        return modelProvider;
    }
}
